JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/slurm/job/JobSubmission.java

```diff
@@ -439,6 +443,17 @@ public void monitorJobs() throws SlurmJobException{
 				session = jsch.getSession(slurmJobProperty.getHpcServerLoginUserName(), getHpcAddress(), 22);
 				String pwd = slurmJobProperty.getHpcServerLoginUserPassword();
 				session.setPassword(pwd);
+
+				try {
+					Connector con = new PageantConnector();
+					IdentityRepository irepo = new RemoteIdentityRepository(con);
+					jsch.setIdentityRepository(irepo);
+					session.setConfig("PreferredAuthentications", "publickey,keyboard-interactive,password");
+				} catch (AgentProxyException e) {
+					session.setConfig("PreferredAuthentications", "password");
+					logger.info("Failed to load identities from Pageant", e);
+				}
```

Comment:
Is there any other way of checking whether there is a valid key present rather than via catching AgentProxyException? Here I am also a little bit concerned about a case where an exception is thrown for a different reason than invalid key potentially. Those are security issues and it is good to be cautious with such things.

Reply:
Also, please confirm that this code is not Windows dependent and will work on *nix like operating systems. Pageant uses win32 APIs, as far as I am aware.

Reply:
The initial call the the PageantConnector constructor will fail quickly if the native Pageant runtime libraries are not found so will default to the password only option if Pageant is not installed/supported on the current system.

Reply:
Regarding your first question, this block of code doesn't actually check any keys, it just tries to connect to a running Pagaent instance and, if successful, registers it with the JSch object. The only exception that can be thrown by any of these methods is AgentProxyException and that is only thrown if the code is unable to load one of the native libraries it requires.

Reply:
You either design it to work differently on different operating systems and make this explicit in your code or find a way to make the functionality being the same on different systems. I'd prefer the second option.